common/modal.js

```diff
@@ -16,6 +16,11 @@
                 if (rejectCB) {
                     rejectCB(response);
                 } else if (typeof response !== "string") {
+                    // there is no internet connection, so templates couldn't be fetched
+                    // set a flag on the response (error) object
+                    if (response.message.includes("Error: [$compile:tpload]")) {
```

Comment:
This could be part of the "catchAll" function. That way we're handling this error type the same way for all the apps. I know the issue was mainly about recordedit, but since by just changing where we're doing this logic we can handle it for all the other apps too, I think we should.

Reply:
This is in `modal.js` as part of the function we use for showing all modals. I figured we only wanted to change the error flow for this case. I'm not sure what other cases would cause `$compile:tpload` error to be thrown, so I didn't want to fully generalize it.

Reply:
Makes sense.